internal/cloudapi/server.go

```diff
@@ -182,7 +182,22 @@ func (server *Server) Compose(w http.ResponseWriter, r *http.Request) {
 		for name, packages := range packageSets {
 			pkgs, _, err := server.rpmMetadata.Depsolve(packages, repositories, distribution.ModulePlatformID(), arch.Name())
 			if err != nil {
-				http.Error(w, fmt.Sprintf("Failed to depsolve base packages for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err), http.StatusInternalServerError)
+				var error_type int
+				switch err.(type) {
+				// DNFException means that the dnf-json tool encountered a non
+				// expected error during its execution. This subtype is handled
+				// as a 500 error.
+				case *rpmmd.DNFException:
```

Comment:
In fact, I'm starting to wonder if we really need the DNFException. Like if json unmarshalling crashed and led to an error, shouldn't it be enough to declare an Internal Server Error ?
Let me now if you prefer that I simplify.

Reply:
Indeed I think it might make more sense to just have DNFError, I agree. And then filtering further based on specific dnf errors.